cmd/nodepool/core/create.go

```diff
@@ -142,3 +144,29 @@ func (o *CreateNodePoolOptions) CreateNodePool(ctx context.Context, platformOpts
 	fmt.Printf("NodePool %s created\n", o.Name)
 	return nil
 }
+
+// ValidateMgmtClusterSupportsNodePoolCPUType validates the HostedCluster payload type can support the CPU architecture
+// of the NodePool.
```

Comment:
please add a unit test for this

Reply:
I'll add one. I was waiting to hear back on https://github.com/openshift/hypershift/pull/4538#discussion_r1764598254.

Reply:
Whoops wrong function